.github/workflows/ai-codereview.yml

```
@@ -86,31 +102,56 @@ jobs:
                 --context "" \
                 --lang "${LANG}" \
                 --template-file "${PATH_TO_JINJA_TEMPLATE_FILE}"
+              set -x  
           done
-          JOB_URL=$(gh run --repo ${{ github.repository }} view ${{ github.run_id }} --json jobs --jq '.jobs[] | select(.name == "${{ github.job }}") | .url, (.steps[] | select(.name == "Run ai-codereview script") | "#step:\(.number):1")' | tr -d "\n")
-          echo "review_result=There is some code review from AI. See the results [here]($JOB_URL)" >> $GITHUB_OUTPUT
+      - name: AI-codereview results
+        id: codereview_results
+        if: steps.changed-golang-files.outputs.any_changed == 'true'
+        env:
+          GH_TOKEN: ${{ github.token }}
+          VENV_DIR_LOC: '.venv'
+        run: |
+          #!/bin/bash -e
+          source ${VENV_DIR_LOC}/bin/activate
+          JOB_URL=$(gh run --repo ${{ github.repository }} view ${{ github.run_id }} --json jobs --jq '.jobs[] | select(.name == "${{ steps.vars.outputs.jobname }}") | .url, (.steps[] | select(.name == "AI-codereview results") | "#step:\(.number):1")' | tr -d "\n")
+          mv output.txt output.md
+          mv summary.txt summary.md
+          rich output.md
+          python3 -m markdown output.md -f output.html
+          echo -e "\nThere is some code review from AI. See the code review [here]($JOB_URL)" >> ./summary.md
+      - name: Upload Artifact
+        id: upload_artifact
+        uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 #v4
+        with:
+          name: my-artifact
```

Maybe change the artifact name for uniqueness and clarity? For example:
```suggestion
          name: ai-codereview-report
```